src/ORM/Table.php

```diff
@@ -2304,4 +2305,28 @@ public function __debugInfo()
             'connectionName' => $conn ? $conn->configName() : null
         ];
     }
+
+    /**
+     * Validates that a belongsToMany or hasMany assoc has at least one entry.
+     *
+     * This can be used to validate any other array as well.
+     *
+     * @param mixed $value
+     * @return boolean
+     */
+    public function validateCount($value, $expectedCount = 0, $operator = '>')
```

Comment:
To add my grain to it, sorry @burzum:

@markstory is right. This touches the Schema/Tables/Associations thus it should be an application rule and not a validation rule.

Reason: This will allow us to move validation rules into entities in future for single/tuple/syntax/entity-comparison field validation. Thus in the tables we can utilize buildRules and in the entity we can utilize validators (hopefully, in future, by default). We should also NOT blur the lines between validation and rules. I remember when this was added late in the 3.x development cycle, and now that we have it, we should utilize and respect the separation.

If we do @lorenzo then we should expose all application rules (such as existsIn and isUnique) as validation methods so that you can run these things in validation context, too.

> What can we do to speed up this process of discussing things without spending *days* or even weeks on getting something done?

I can only agree with this and I feel for you, but as with you most (all?) guys here do contribute without being employed to do so. So stuff takes time.